### ISSUE TITLE
Not booting after running out of battery.

### ISSUE THREAD

**Issue by Casm101** [daliansky/XiaoMi-Pro-Hackintosh#661]
I have an issue, that normally occurs when leaving my computer to go into sleep mode, and forgetting about it until it runs out of battery... (This happens often, every single time I have run down the battery while having MacOS and OpenCore).

As soon as I plug it back in, and attempt to start it, the indicator light turns on, but the screen stays black, no other activity is visible on the laptop. I have attempted fully charging my device, then turning it on, but the same issue occurs. 

To fix this issue, I have to open up the laptop, remove the boot drive, attempt to boot a couple of times, until the system turns on, displays no boot drive available, I put the drive back in, then I am able to start the system as usual. It's as if the system gets stuck in the sleep / hibernate mode.

Has anyone had this issue before? Any recommendations as how to solve it?

**Comment by stevezhengshiqi** [comment]
I also had this issue last time. The first thought is to remove HibernationFixup, and change `Misc -> Boot -> HibernateMode -> None`. I am not sure why the issue could happen. A helpful resource would be https://dortania.github.io/OpenCore-Post-Install/universal/sleep.html#fixing-sleep

**Comment by Casm101** [comment]
I removed the kext, and changed the option in my config file, ran down the battery, and as soon as I plugged it in, it booted first try! 

This raises a question for me though, when I close the lid, or put the system to sleep, is it actually sleeping? Or is it still on with screen off? Since when I have woken it, it hasn't asked me for my password to log back in, which isn't ideal as I use my laptop for work, and rather not have people being able to snoop my work.

**Comment by stevezhengshiqi** [comment]
> as soon as I plugged it in, it booted first try!

Sounds perfect! Could you also try adding back HibernationFixup.kext since it helps save battery life by enabling hibernation mode. This time just change Misc -> Boot -> HibernateMode -> None in config.

> it hasn't asked me for my password to log back in

You can try System Preferences - Security & Privacy - Require Password - Immediate

**Comment by stevezhengshiqi** [comment]
I actually feel that adding back HibernationFixup.kext would cause the no boot problem. In my understanding, then the device was low in battery and hibernates, the low battery protection (Force shutdown when battery low) was not guarding for some reason, and resulted in our device draining too much battery in hibernation mode and can not boot. Need actual tests to verify.

**Comment by Casm101** [comment]
OKay, so I let the laptop run flat, with the kext installed, and the hibernation mode set to none, and it wont boot, so I guess the solution for now is to leave it without the hibernationfix kext...

**Comment by stevezhengshiqi** [comment]
Thank you for testing that! I believe it would be a pain to recover from that. I will disable HibernationFixup.kext for 10th Gen U device in the next release.

**Comment by Casm101** [comment]
Yeah, was a bit of a pain to get running again, had to use an Ubuntu key to boot, hotplug the M.2, mount the EFI and delete the kext..., thanks for the help @stevezhengshiqi

**Comment by toxa70x4** [comment]
-hbfx-disable-patch-pci disables patching of IOPCIFamily (this patch helps to avoid hang & black screen after resume (restoreMachineState won't be called for all devices)

**Comment by stevezhengshiqi** [comment]
@toxa70x4 Our issue is laptop won't be boot up after we left it hibernate for a long time when battery low. I don't see much connection with -hbfx-disable-patch-pci. When battery level is higher than critical level, the laptop hibernates perfect with HibernationFixup.

**Comment by toxa70x4** [comment]
I don't know how much battery power you need. 8% is not a problem.
I have a TM1963 laptop.
Parameter -hbfx-disable-patch-pci, kext NVMeFix and slightly different SSDT-USB.aml

**Comment by stevezhengshiqi** [comment]
@toxa70x4 Did you meet the exact same problem as us before? The laptop can not boot even you charged it to full battery.

what happened to us is that we hibernates our laptop when battery is around 5%, and leave it there one day or even one week. Then it is unbootable even connected with power.

-hbfx-disable-patch-pci I think it is not recommended to have. As the deacription says, the pci patch can fix blacl screen, and adding this boot argument will disable the fix.

**Comment by stevezhengshiqi** [comment]
@Casm101 Hi, in the commit https://github.com/daliansky/XiaoMi-Pro-Hackintosh/commit/b902144e36f9b0af9aeb420d5fbf854fcbfecd7b, I re-enabled HibernationFixup; and if you still need to disable hibernation to avoid this problem, you can follow https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/main/Docs/README_CML.txt

**Comment by Casm101** [comment]
@stevezhengshiqi thanks for the update, I'll check it out when I transition to the latest version
